packages/runtime/container-runtime/src/dataStoreContext.ts

```diff
@@ -676,7 +681,9 @@ export abstract class FluidDataStoreContext extends TypedEventEmitter<IFluidData
      * This method should not be used outside of the aliasing context.
      * It will be removed, as the source of truth for this flag will be the aliasing blob.
      */
-    public abstract setRoot(): void;
+     public setRoot(): void {
```

Comment:
Is this API is called only as result of aliasing? If so, maybe we should rename it to reflect it.

Reply:
No, it's not

Reply:
Where else is it used?

Reply:
Based on the comment below, it should be used when a local datastore is created

Reply:
Okay. I was looking at existing usages. You can still call this `setAliased` or something like that. Like I mentioned in my other comment, this isn't exactly true since being root doesn't mean its aliased but its okay because it makes the code simpler and easier to reason about.

Reply:
The problem is there are 3 states that need to be handled:
1. root during creation.
2. Aliased.
3. root in initial snapshot.
For simplicity, we can combine 1 and 2. Also, local data stores only care about 1 and 2.
Alternatively, all 3 states can be tracked. Local data stores look at 1 and 2. Remote data stores look at all 3.